source/grid/tria.cc

```diff
@@ -7170,19 +7030,34 @@ namespace internal
                                    face->vertex_index(1),
                                    face->vertex_index(2)}};
 
-                                const auto &new_hex_vertices =
-                                  cell_vertices_tet[chosen_line_tetrahedron][c];
-                                const auto face_f_vertices =
-                                  reference_cell_face_vertices[f];
-
+                                // the 8 child tets are each defined by 4
+                                // vertices the ordering of the tets has to be
+                                // consistent with above the ordering within the
+                                // tets is given by the reference tet i.e.
+                                // looking at the fifth line the first 3
+                                // vertices are given by face 11, the last
+                                // vertex is the remaining of the tet
+                                const auto new_hex_vertices =
+                                  hex->reference_cell()
+                                    .new_isotropic_child_cell_vertices(
+                                      chosen_line_tetrahedron)[c];
+
+                                // arrange after vertices of the faces of the
+                                // unit cell
                                 const std::array<unsigned int, 3> vertices_1 = {
                                   {
                                     vertex_indices
-                                      [new_hex_vertices[face_f_vertices[0]]],
+                                      [new_hex_vertices
+                                         [hex->reference_cell()
+                                            .face_to_cell_vertices(f, 0, 1)]],
                                     vertex_indices
-                                      [new_hex_vertices[face_f_vertices[1]]],
+                                      [new_hex_vertices
+                                         [hex->reference_cell()
+                                            .face_to_cell_vertices(f, 1, 1)]],
                                     vertex_indices
-                                      [new_hex_vertices[face_f_vertices[2]]],
+                                      [new_hex_vertices
+                                         [hex->reference_cell()
+                                            .face_to_cell_vertices(f, 2, 1)]],
```

Comment:
If I read the warning message at https://cdash.dealii.org/viewBuildError.php?type=1&buildid=3972 correctly (see #16949), this is the offending line. What happens is that we pass `2` as a constant into `face_to_cell_vertices`, but that function does not know about the fact that the reference cell is in 3D, so it looks up `GeometryInfo<2>::face_to_cell_vertices`, which then calls `GeometryInfo<2>::child_cell_on_face` with an index `subface` out of range (we give it 3 but it gets 2). @dominiktassilostill how should we proceed? In the end, we only want the tetrahedron to tell us the numbering, right?

Reply:
Yes, here we only need the information of the tetrahedron, it is not called in the case of the hexahedron. Would it be enought to change line 7014 `if (new_hex->n_faces() == 4)` to `if (reference_cell_type == ReferenceCells::Tetrahedron)`?

Reply:
One can never be entirely sure with a compiler without explicitly trying it out, but I think it is worth a try. I am not sure how easy it would be to create a test code for use with [godbolt.org](https://godbolt.org/) which would allow you to test the precise compiler - one can save the temporary the compiler generates or try to come up with a small case.